### PR TITLE
[ARV] Confirm reporting blocked users

### LINKED ISSUE #816: Warn if reporting a blocked user to AIV/UAA
It would be helpful if Twinkle popped up a warning if you try to report a user to UAA or AIV when they're already blocked - I've made that mistake a few times. It's not a critical issue, since I'm pretty sure a bot will come along and clean up the filing, but it would be nice to have.

### PR DESCRIPTION
**Pull request by DannyS712**
Before proceeding to open the form, check if the relevant user is blocked or range blocked. If blocked, request confirmation.
The former "Twinkle.arv.callback" is now at "Twinkle.arv.callback.showform" - other than the portletLink, I didn't see any callers
Closes #816

### CODE REVIEW

**Comment by DannyS712** [comment]
Testing in developer tools:
* Delete the existing portlet
* Paste in the new code
* Call `Twinkle.load()`
Tested with user blocked, IP blocked, range blocked, and no blocks, works

File: modules/twinklearv.js [hunk 1]
```diff
@@ -18,14 +18,57 @@ Twinkle.arv = function twinklearv() {
...
+		list: 'blocks',
+		bkprop: 'range'
+	};
+	if (isIP) {
+		params.bkip = username;
+	} else {
+		params.bkusers = username;
+	}
+
+	new mw.Api().get(params).then(function(data) {
```
Comment (Amorymeltzer) [inline modules/twinklearv.js]: Presumably this would be nicer after #765 ?

Comment (DannyS712) [reply modules/twinklearv.js]: Not sure - this is based on code in checking for blocks elsewhere in the repo (don't remember where exactly) but morebits api may not be needed

Comment (Amorymeltzer) [reply modules/twinklearv.js]: [twinkleblock](https://github.com/azatoth/twinkle/blob/f1135f6304300eb55624e38ece6e921e209afc4b/modules/twinkleblock.js#L98-L107).  No, it's not needed, but I believe it's the kind of thing that #765 would enable `Morebits.wiki.api` to handle (note: I haven't really reviewed #765).  Not necessarily suggesting you change it, just noting its existence, especially given that it's an odd-one-out.

File: modules/twinklearv.js [hunk 2]
```diff
@@ -18,14 +18,57 @@ Twinkle.arv = function twinklearv() {
...
 	}, 'ARV', 'tw-arv', title);
 };
 
-Twinkle.arv.callback = function (uid) {
+Twinkle.arv.callback = function (username, isIP) {
+	var params = {
+		format: 'json',
+		action: 'query',
+		list: 'blocks',
+		bkprop: 'range'
```
Comment (Amorymeltzer) [inline modules/twinklearv.js]: Probably only needs to be added if `isIP`, right?

**Comment by Amorymeltzer** [comment]
Thoughts on the above, @DannyS712?

Beyond that, do we really want to check for every option before loading?  Notifying before reporting to AIV and UAA is good, they'll just get autoremoved otherwise.  I guess AN3 as well, but do we definitely want to do this before SPI reporting?  Especially for the sock option.  I suppose it doesn't hurt much, but seems a waste, especially if it's going to be delaying opening the window.

Just thinking out loud: what about checking after the user has submitted the form?  We could cancel if AIV/UAA, then check/notify for AN3/SPI.  Does that make sense or am I mixed up?

File: modules/twinklearv.js (hunk 2 above)
Comment (DannyS712) [reply modules/twinklearv.js]: from what I recall, this is ignored if its not an IP

**Comment by siddharthvp** [comment]
I too don't think it's good practise to put an API call that delays the opening of the form. One expects the something to open immediately when the they click the menu button. Using the red text option (as in #757) would be better IMO. 

Also, is there a reason why `mw.Api` is used instead of morebits.wiki.api? The latter now supports JSON, so you can simply replace 
```new mw.Api().get(params).then(successFunc, failureFunc)``` with
```new Morebits.wiki.api('Checking if already blocked', params, successFunc, null, failureFunc).post()```

**Comment by Amorymeltzer** [comment]
Done in #1184

**Pull request closed without merging**